cpp/src/request_am.cpp

```diff
@@ -119,15 +199,34 @@ ucs_status_t RequestAm::recvCallback(void* arg,
   bool is_rndv = param->recv_attr & UCP_AM_RECV_ATTR_FLAG_RNDV;
 
   std::shared_ptr<Buffer> buf{nullptr};
-  auto allocatorType = *static_cast<const ucs_memory_type_t*>(header);
+  // auto amHeader         = AmHeader::deserialize(static_cast<const char*>(header));
+  auto amHeader =
+    AmHeader::deserialize(std::string(static_cast<const char*>(header), header_length));
```

Comment:
This copies the header, *suggestion*:
```suggestion
    AmHeader::deserialize(std::string_view(static_cast<const char*>(header), header_length));
```
And accept a `std::string_view` in `deserialize`.

Reply:
Done in https://github.com/rapidsai/ucxx/pull/186/commits/4304378c4b2ea578d97985782843a1bc04a405e0 .